lib/encoding_dot_com/format.rb

```diff
@@ -74,6 +74,15 @@ def build_node(builder, node, attributes)
             value = output_value(key, value)
             if value.kind_of?(Hash)
               build_node builder, key, value
+            # if the value is an array, loop through the array and process each item based on its type
+            elsif value.kind_of?(Array)
+              value.each do |v|
+                if v.kind_of?(Hash) || v.kind_of?(Array)
+                  build_node builder, key, v
+                else
+                  builder.send("#{key}_", v) unless v.nil?
```

Comment:
Instead of having `unless v.nil?` here, I would add a guard right before the `if` statement.
Basically adding `next if v.empty?` or `next unless v.present?` before if block.
Using `.empty?` will catch any empty value.

Reply:
Normally, yes. 😄 

Our end goal is to be able to pass in two identical, self-closing XML nodes:
```
<format>
  <audio_stream/>
  <audio_stream/>
</format>
```

This gem made assumptions on how to build the XML output with Nokogiri by building a hash. This leads to a problem where we can't have, say, `{'audio_stream' => '1', 'audio_stream' => '2'}` in our attributes when we build the XML, for example.

In our specific use case, we need to be able to build an XML file with multiple audio streams that don't have any other attributes attached to them. What this code allows is for us to send in an array as the value to a key. When we do this, we are specifically creating `x` nodes with the same name as the key where `x` is the number of items in the array. If we want those nodes to be self-closing, we pass in an empty string for that item in the array.

Additionally, the array can hold a hash as well in the event that we want to add child nodes.

The specs we wrote should (hopefully) illustrate what's going on. The code is definitely confusing at a glance. 😆